product-backend/src/main/java/org/kainos/ea/validator/RegisterValidator.java

```diff
@@ -0,0 +1,50 @@
+package org.kainos.ea.validator;
+
+import org.kainos.ea.cli.RequestEmployee;
+
+public class RegisterValidator {
+        // Minimum password length (you can adjust this as needed)
+        private static final int minLength = 8;
+
+        public static boolean validate (RequestEmployee input) throws Exception {
+            String email = input.getEmail();
+            String password = input.getPassword();
+            String role = input.getRole();
+
+            if (!role.equals("Admin") && !role.equals("Employee")) {
+                throw new Exception("Correct role should be specified.");
+            }
+
+            if (email.contains(" ")) {
```

Comment:
You should validate email is under @kainos.com domain

```suggestion
            if (email.endsWith("@kainos.com")) {
```

Reply:
@kainos.com check yes, and something that checks the email is not empty and as above does not contain whitespace (usually we do a trim() of a string and check if it's an empty value so that even tabulators are checked.